Gaming The Syllabus/Assets/Scripts/LevelLoader.cs

```diff
@@ -19,34 +20,33 @@ void Update()
             LoadNextLevel();
         }
         */
-        if(GameObject.Find("Portal").transform.position.y <=GameObject.Find("Player").transform.position.y ){
-            LoadNextLevel();
-        
-            
-            
+        if(transform.position == GameObject.Find("Player").transform.position) {
```

Comment:
GameObject.Find("Player") in Update is extremely intensive. Let's try calling this once in Start and storing the result in a variable to use later.

Reply:
Does it not need to be checked constantly?

Reply:
Once you have a reference to the player GameObject you can use player.transform.position which will be the current position

Reply:
`GameObject player = GameObject.Find("Player");
if (transform.position == player.transform.position) {
    . . .
}`